CONTRIBUTING.rst

```diff
@@ -0,0 +1,313 @@
+======================
+Contributing to pyiron
+======================
+
+The following is a set of guidelines for contributing to pyiron, which is
+hosted and maintained by the `Max Planck Institut für Eisenforschung`_
+on GitHub. These are mostly guidelines to facilitate an efficient
+development workflow, and not necessarily rules. Use your best judgment,
+and feel free to propose changes even to this document in a pull request.
+
+You can find all the pyiron packages at our `github page`_ .
+To create pull requests, you will need to become part of the
+pyiron organization. Please email us if you would like to join.
+
+Wait I don't want to read this; I just have a quick question/bugfix!
+====================================================================
+
+1. Check out our `FAQ page`_; your question might already be answered there.
+2. If your question relates to a bug in pyiron, please briefly search the `issues page`_ and open a new labeled issue if you don't see anything related to your question there.
+3. Please feel free just to send one of us a brief, descriptive email with your question, and we'll do our best to get back to you as ASAP as possible.
+
+Table of Contents
+=================
+
+`Code of conduct`_
+
+`What should I know before I get started?`_
+  * `The structure of pyiron`_
+  * `The principles of pyiron`_
+
+`How can I contribute?`_
+  * `Reporting bugs`_
+  * `Suggesting enhancements`_
+  * `Your first code contribution`_
+  * `Pull requests`_
+
+`Styleguides`_
+  * `Git commit messages`_
+  * `Python styleguide`_
+  * `Documentation styleguide`_
+
+`Additional Notes`_
+  * `Issue and pull request labels`_
+  * `Build status`_
+  * `pyiron releases`_
+
+Code of conduct
+===============
+
+This project and everyone participating in it is governed by
+the pyiron `Code of conduct`_. By participating, you are expected
+to uphold this code. Please email us to report violations.
+
+.. _What should I know before I get started?:
+What should I know before I get started?
+========================================
+
+The structure of pyiron
+-----------------------
+
+The principles of pyiron
+------------------------
+
```

Comment:
I think the structure & principles of pyiron is something that will change drastically in the future based on the discussions we had earlier. For the time being, we could save these lines as a comment and work on it once the new structure is finalized.

Reply:
> This looks great! Should we also have a markdown version `CONTRIBUTING.md` on github?

I'm happy to also make a markdown version, but it seems the rst file also renders automatically online. Everything else was in rst for the sphinx docs, so that's why I left it that way.

> I think the structure & principles of pyiron is something that will change drastically in the future based on the discussions we had earlier. For the time being, we could save these lines as a comment and work on it once the new structure is finalized.

sounds good 👍